.github/workflows/build.yaml

```
@@ -0,0 +1,100 @@
+name: build
+on: [push, pull_request]
+jobs:
+  distcheck:
+    name: "distcheck ${{ matrix.id }}"
+    runs-on: ubuntu-latest
+    container: debian:stable
+    timeout-minutes: 360
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - { id: "debian-stable-amd64-gcc",   compiler: { CC: "/usr/bin/gcc",   CXX: "/usr/bin/g++",     pkg: "gcc g++" } }
+          - { id: "debian-stable-amd64-clang", compiler: { CC: "/usr/bin/clang", CXX: "/usr/bin/clang++", pkg: "clang"   } }
+    env:
+      CC: "${{ matrix.compiler.CC }}"
+      CXX: "${{ matrix.compiler.CXX }}"
+    steps:
+      - name: Get number of CPU cores
+        run: echo "NUM_CPU_CORES=$(grep -c processor /proc/cpuinfo)" >> $GITHUB_OUTPUT
+      - name: Update package database
+        run: apt-get update -y
+      # See comments beginning at
+      # https://github.com/actions/runner/issues/763#issuecomment-1435474884
+      # Without Git, actions/checkout@v3 will resort to REST and will not
+      # create a .git folder or .git.config. The Problem Matcher looks for
+      # .git/config to find where the root of the repo is, so it must be
+      # present.
+      - name: Install Git
+        run: apt-get -y install git
+      - uses: actions/checkout@v3
+      - name: Install build tools
+        shell: bash
+        run: |
+          apt-get -y install pkg-config libtool autoconf \
+            automake g++ bison flex make bash-completion dh-autoreconf \
+            debhelper devscripts wget python3-pip
+      - name: Install build dependencies
+        shell: bash
+        run: |
+          apt-get -y install libcppunit-dev uuid-dev libncurses5-dev \
+            libmicrohttpd-dev protobuf-compiler python3-protobuf \
+            libprotobuf-dev libprotoc-dev zlib1g-dev libftdi-dev \
+            libusb-1.0-0-dev liblo-dev libavahi-client-dev python3-numpy \
+            default-jdk-headless maven
+      - name: Install compiler
+        shell: bash
+        run: apt-get -y install ${{ matrix.compiler.pkg }}
+      - name: Autoreconf
+        run: autoreconf -i
+      - name: Set configure arguments
+        run: |
+          echo "GH_OLA_CONFIGURE_ARGS=--enable-rdm-tests --enable-java-libs --enable-ja-rule --enable-e133" >> $GITHUB_ENV
+      - name: Set additional Linux configure arguments
+        if: runner.os == 'Linux'
+        # Silence all deprecated declarations on Linux due to auto_ptr making the build log too long
+        run: |
+          echo "GH_OLA_CONFIGURE_ARGS=$GH_OLA_CONFIGURE_ARGS CPPFLAGS=-Wno-deprecated-declarations" >> $GITHUB_ENV
+      - name: Print configure command
+        run: echo "./configure $GH_OLA_CONFIGURE_ARGS"
+      - name: Configure
+        run: ./configure $GH_OLA_CONFIGURE_ARGS
+      - name: distcheck
+        run: make distcheck -j${{ steps.num-cpu-cores.outputs.NUM_CPU_CORES }} VERBOSE=1
```

So have you tried this? Getting us just `make check` for now would be great if that is quick enough to not time out?
```suggestion
      - name: makecheck
        run: make check -j${{ steps.num-cpu-cores.outputs.NUM_CPU_CORES }} VERBOSE=1
```

There's also ccache which we've used successfully in the past and ought to speed things up if the cache is cached too.

---

I'll try simply `make check`!

What's weird is that it isn't timing out per say...it's almost like a watchdog has decided there hasn't been any output for a while and that is why it is stopping or some OOM situation. I think I need to open an issue elsewhere about it.

Yeah I'll look into ccache.